pkg/controller/ipam/validate.go

```diff
@@ -117,141 +86,77 @@ func validationResult(allowed bool, msg string) *admv1.AdmissionResponse {
 	}
 }
 
-// getIPRangeDifference returns SubnetIPRanges that are in s1 but not in s2.
-func getIPRangeDifference(s1, s2 []crdv1beta1.IPRange) []crdv1beta1.IPRange {
-	newSet := map[crdv1beta1.IPRange]struct{}{}
-	for _, ipRange := range s2 {
-		newSet[ipRange] = struct{}{}
-	}
+func validateIPRangesAndSubnetInfo(ipPool crdv1beta1.IPPool) (string, bool) {
```

Comment:
can we unify further by having a shared function taking an `[]IPRange` and a `*SubnetInfo` as parameters? Isn't it correct that the only difference between `ExternalIPPool` validation and `IPPool` validation is the check for overlapping IP ranges for `ExternalIPPool` (which we don't have for `IPPool`)?

Reply:
done